samples/apps/gradio_gui/plugin.py

```diff
@@ -0,0 +1,147 @@
+from void_terminal.toolbox import ChatBotWithCookies, update_ui
+from void_terminal.toolbox import get_conf, select_api_key
+from void_terminal.toolbox import Singleton
+
+
+@Singleton
+class GradioMultiuserManagerForPersistentClasses:
+    """
+    A class for managing multiple instances of persistent classes as singletons.
+
+    This class allows storing and accessing multiple instances of persistent classes
+    and ensures that only one instance exists for each class. It provides methods for
+    checking if an instance with a certain key already exists, setting a new instance,
+    and getting an existing instance.
+
+    Attributes:
+        mapping (dict): A dictionary to store the mapping between keys and instances.
+    """
+
+    def __init__(self):
+        self.mapping = {}
+
+    def already_alive(self, key):
+        return (key in self.mapping) and (self.mapping[key].is_alive())
+
+    def set(self, key, x):
+        self.mapping[key] = x
+        return self.mapping[key]
+
+    def get(self, key):
+        return self.mapping[key]
+
+
+def autogen_terminal(
+    txt,
+    llm_kwargs: dict,
+    plugin_kwargs: dict,
+    chatbot: ChatBotWithCookies,
+    history: list,
+    system_prompt: str,
+    web_port,
+    AutoGenFn: callable,
+    Callback: str,
+):
+    """
+    Function for handling inputs and executing autogen_terminal module.
+
+    Parameters:
+        txt (str): Text entered by the user in the input field, for example, a paragraph to be translated or a file path containing files to be processed
+        llm_kwargs (dict): Parameters for the gpt model, such as temperature and top_p, typically passed as is
+        plugin_kwargs (dict): Parameters for the plugin model
+        chatbot (object): Handle for the chat display box, used to display output to the user
+        history (object): Chat history, context
+        system_prompt (str): Silent reminder for the gpt
+        web_port (int): Port number on which the software is running
+        AutoGenFn (function): Function for autogeneration
+        Callback (function): Callback function for plugin
+
+    Returns:
+        None
+    """
+    # Check if the current model meets the requirements
+    supported_llms = [
+        "gpt-3.5-turbo-16k",
+        "gpt-4",
+        "gpt-4-32k",
+        "azure-gpt-3.5-turbo-16k",
+        "azure-gpt-4",
+        "azure-gpt-4-32k",
+    ]
```

Comment:
Model list in the UI is not the same as listed here. Is the list in UI defined in `gradio-forked`?
More important thing is that we need to allow custom value here, or the model specified in `OAI_CONFIG_LIST` should be added automatically.
![image](https://github.com/microsoft/autogen/assets/3197038/c9c9f78e-165e-448b-824b-316be9dcbc19)

Reply:
This dropdown is for direct chat (instead of AutoGen), therefore it can support more models (AutoGen only use OAI_CONFIG_LIST). As far as I'm concerned, some of these should indeed be removed.

Furthermore, in the main resp of GPT-Academic, we have successfully hook the model callback to all llm interfaces, e.g. chatglm3, claude, claude2, internlm, qwen, etc. However, the hook code seems not robust and generic enough (https://github.com/binary-husky/gpt_academic/blob/c4aefc5fac7a21e0c4743ae16a5a71790832e5f8/crazy_functions/agent_fns/general.py#L81C47-L81C47), we would be happen to hear from your oppions

Reply:
I'll run more test about this menu later

Reply:
> This dropdown is for direct chat (instead of AutoGen), therefore it can support more models (AutoGen only use OAI_CONFIG_LIST). As far as I'm concerned, some of these should indeed be removed.
> 
> Furthermore, in the main resp of GPT-Academic, we have successfully hook the model callback to all llm interfaces, e.g. chatglm3, claude, claude2, internlm, qwen, etc. However, the hook code seems not robust and generic enough (https://github.com/binary-husky/gpt_academic/blob/c4aefc5fac7a21e0c4743ae16a5a71790832e5f8/crazy_functions/agent_fns/general.py#L81C47-L81C47), we would be happen to hear from your oppions

I'm confused. I uploaded a good configure with `OAI_CONFIG_LIST`, but it raised an error saying the config is not valid when I use direct chat. How can I set up LLM config for direct chat, I think it should be the same as agent chat. And, what should I do to trigger an agent chat?